src/components/utils/src/threads/thread_qt.cc

```diff
@@ -113,11 +116,14 @@ bool Thread::start() {
 }
 
 uint64_t Thread::CurrentId() {
-  return QThread::currentThread();
+  // Available interfaces QThread::currentThread() and
+  // QThread::currentThreadId() return values, which can't be converted to
+  // uint64_t, thus using native WIN interface.
+  return static_cast<uint64_t>(GetCurrentThreadId());
 }
 
 bool Thread::IsCurrentThread() const {
-  return CurrentId() == thread_handle();
+  return QThread::currentThread() == thread_handle();
```

Comment:
@anosach-luxoft 
I see that conversion to uint64_t is necessary only once - in CurrentId(), so I would prefer to use native QT handle instead of converted value.